### PR TITLE
[UWP] Add data uri for images support

### PR DESCRIPTION
**Pull request by almedina-ms**
Solves this issue: https://github.com/Microsoft/AdaptiveCards/issues/2128

### CODE REVIEW

File: source/shared/cpp/ObjectModel/AdaptiveBase64Util.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    size_t enc_len = 0;
+    unsigned char a3[3];
+    unsigned char a4[4];
+
+    out->resize(EncodedLength(in));
+
+    size_t input_len = in.size();
+    std::string::const_iterator input = in.begin();
+
+    while (input_len--) {
```
Comment (khouzam) [inline source/shared/cpp/ObjectModel/AdaptiveBase64Util.cpp]: Needs to follow the clang-format

File: source/shared/cpp/ObjectModel/AdaptiveBase64Util.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+std::string AdaptiveBase64Util::Encode(const std::string& decodedBase64)
+{
+    std::string encodedString;
+    Encode(decodedBase64, &encodedString);
+    return encodedString;
+}
+
+// Format for DataURI is data:[<MediaType>][;base64],data with MediaType and base64 being optional and data is composed of [A-Z a-z 0-9 + /] characters
+std::string AdaptiveBase64Util::ExtractDataFromUri(const std::string& dataUri)
```
Comment (khouzam) [inline source/shared/cpp/ObjectModel/AdaptiveBase64Util.cpp]: >std::string [](start = 0, length = 11)

This should return a vector<char> or similar

File: source/shared/cpp/ObjectModel/AdaptiveBase64Util.cpp [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+    if (base64Alphabet.find(c) == base64Alphabet.end())
+    {
+        return 255;
+    }
+    return base64Alphabet[c];
+}
+
+bool AdaptiveBase64Util::Encode(const std::vector<char>& in, std::string* out)
+{
```
Comment (khouzam) [inline source/shared/cpp/ObjectModel/AdaptiveBase64Util.cpp]: Can we have a some tests in the shared model to cover this code?

**Comment by khouzam** [comment]
One more minor thing. Can you move the files to v1.1. We're going to backport this because Timeline needs it in Windows.

**Comment by bandmask** [comment]
@almedina-ms @khouzam 
Currently having problem with the functionality this is intended to solve.
We cannot render data uris in format of `data:image/png:base64 [base64EncodedString]` without implementing our own image renderer (converting the AdaptiveImage to and UWP Image with source as BitmapImage).

I would like to request some enlighting pointers for how to do this without using a custom renderer.

We have tried images both as a direct child to the adaptiveCard body, and as a child element in a nested hierarchy, without any success.

Since the #2128 along with #628 both are closed, one would think this should work out of the box?

Best regards

**Pull request merged**